swarm/network/stream/snapshot_sync_test.go

```diff
@@ -0,0 +1,568 @@
+			log.Debug(fmt.Sprintf("subscribe: %d", j))
+			ctx, cancel := context.WithTimeout(ctx, 1*time.Second)
+			defer cancel()
+			client, err := net.GetNode(ids[j]).Client()
+			if err != nil {
+				return err
+			}
+			err = client.CallContext(ctx, nil, "stream_startSyncing")
+			if err != nil {
+				log.Error(fmt.Sprintf("FAILED CallContext %v", err))
+				return nil
+			}
+		}
+		time.Sleep(10 * time.Second)
+		//now upload the chunks to the selected random single node
+		conf.chunks, err = uploadFileToSingleNodeStore(node.ID(), chunkCount)
+		if err != nil {
+			return err
+		}
+		//finally map chunks to the closest addresses
+		conf = mapKeysToNodes(conf)
+		log.Debug(fmt.Sprintf("%v", conf.nodesToChunksMap))
+
+		return nil
+	}
+
+	//for each tick, run the checks on all nodes
+	go func() {
+		ticker := time.NewTicker(time.Second * 1)
+		for range ticker.C {
+			for i := 0; i < len(ids); i++ {
+				log.Debug(fmt.Sprintf("triggering step %d, id %s", i, ids[i]))
+				trigger <- ids[i]
+			}
+		}
+	}()
+	/*
+		go func() {
+			startTime = time.Now()
+			ticker := time.NewTicker(time.Second / 10)
+			for range ticker.C {
+				checkChunkIsAtNode(conf)
+			}
+		}()
+	*/
+
+	//run the simulation
+	result := simulations.NewSimulation(net).Run(ctx, &simulations.Step{
+		Action:  action,
+		Trigger: trigger,
+		Expect: &simulations.Expectation{
+			Nodes: ids,
+			Check: triggerCheck,
+		},
+	})
+	//close(quitC)
+	if result.Error != nil {
+		return result.Error
+	}
+
+	return nil
+}
+
+func (r *TestRegistry) StartSyncing(ctx context.Context) error {
+	var err error
+
+	add := r.addr.ID()
+	pp := ppmap[add]
+	h := r.delivery.overlay.Healthy(pp)
+	fmt.Println("----------------------------------")
+	fmt.Println(r.delivery.overlay.String())
+	fmt.Println(fmt.Sprintf("IS HEALTHY: %t", h.GotNN && h.KnowNN && h.Full))
+
+	pos := make(map[int]discover.NodeID)
+
+	r.delivery.overlay.EachConn(nil, 256, func(addr network.OverlayConn, po int, nn bool) bool {
+		lastPO := po
+		if nn {
+			lastPO = maxPO
+		}
+		peerId := conf.addrToIdMap[string(addr.Address())]
+		fmt.Println(fmt.Sprintf("node %s has conn with %s at po %d and is nn: %t", r.addr.ID(), peerId, po, nn))
+		pos[po] = peerId
+		for i := po; i <= lastPO; i++ {
+			err = r.Subscribe(peerId, NewStream("SYNC", []byte{byte(i)}, false), &Range{From: 0, To: 0}, Top)
+			if err != nil {
+				log.Error(fmt.Sprintf("Error subscribing! %v", err))
+				return false
+			}
+		}
+		return true
+	})
+	prev := 0
+	kad, ok := r.delivery.overlay.(*network.Kademlia)
+	if !ok {
+		return fmt.Errorf("Not a Kademlia!")
+	}
+
+	kad.EachBin(r.addr.Over(), pof, 0, func(po, size int, f func(func(val pot.Val, i int) bool) bool) bool {
+		skip := po - prev
+		/*
+			fmt.Println(prev)
+			fmt.Println(po)
+			fmt.Println(skip)
+		*/
+		remember := make(map[int]bool)
+		if skip > 1 {
+			f(func(val pot.Val, i int) bool {
+				//for c := po + 1; c < po+skip; c++ {
+				for c := po - 1; c > po-skip; c-- {
```

Comment:
@zelig this will go backwards from a po over empty bins. So if we have 
```
000 fd34
001 ef66
002 0
003 aa4d
```
it will do the `RequestSubscription` on the peer from `003` bin - is this correct? or should it rather be  the other way round, to peers from bin `001`?